src/components/measurements/measurementsD3.js

```diff
@@ -223,7 +230,57 @@ export const drawMeasurementsSVG = (ref, svgData) => {
 export const colorMeasurementsSVG = (ref, treeStrainColors) => {
   const svg = select(ref);
   svg.selectAll(`.${classes.rawMeasurements}`)
-    .style("fill", (d) => treeStrainColors[d.strain]);
+    .style("fill", (d) => treeStrainColors[d.strain].color);
+};
+
+export const drawMeansForColorBy = (ref, svgData, treeStrainColors) => {
+  const { xScale, groupedMeasurements } = svgData;
+  const svg = select(ref);
+  // Re move all current color by means
+  svg.selectAll(`.${classes.colorMean}`).remove();
+  // Calc and draw color by means for each group
+  groupedMeasurements.forEach(([groupingValue, measurements]) => {
+    // For each color-by attribute, create an array of measurement values and keep track of color
```

Comment:
While playing around with the review app, I realized this doesn't work well for color-by's with a continuous scale.  I think I have to group by the legend values rather than the specific strain attribute values.

Reply:
Nice find. The legend is the only (?) place which performs this binning, but I've wanted to use this binning to create a set of normal auspice filters, which I think is exactly the approach you're proposing for the group-by here. Currently we ignore continuous traits for the filters, and that approach would be perfectly acceptable for this PR if you don't have time to use the legend binning.

Reply:
Resolved by 666d8b994f6fc83483cd672c451a681d5ebf458c